test/e2e/e2e-utils.ts

```diff
@@ -0,0 +1,96 @@
+import zlib from 'zlib';
+import _ from 'lodash';
+import path from 'path';
+import { Definition } from 'nock';
+
+type Recording = Definition & { rawHeaders: string[] }; // Hack to fix issue in Nock types that does not include `rawHeaders` property in `Definition` type
+
+export function decodeBuffer(recordingResponse: Definition['response']) {
+  try {
+    // Decode the hex buffer that nock made
+    const response = Array.isArray(recordingResponse)
+      ? recordingResponse.join('')
+      : recordingResponse?.toString() || '';
+    const decoded = Buffer.from(response, 'hex');
+    const unzipped = zlib.gunzipSync(decoded).toString('utf-8');
+    return JSON.parse(unzipped);
+  } catch (err) {
+    throw new Error(`Error decoding nock hex:\n${err}`);
+  }
+}
+
+export function afterRecord(recordings: Definition[]): Definition[] {
+  return recordings.map((recording) => {
+    recording.response = decodeBuffer(recording.response as string);
+    return sanitizeRecording(recording);
+  });
+}
+
+export function testNameToFilename(testName = ''): string {
+  if (testName === '') throw new Error('Test name not defined');
+  return `${testName.replaceAll(' ', '-')}.json`;
+}
+
+export function testNameToWorkingDirectory(testName = ''): string {
+  const directoryName = testName.replaceAll(' ', '-');
+  return path.join('./local/recorded', directoryName);
+}
+
+export function sanitizeRecording(recording: Definition): Definition {
+  const sanitizedRecording = recording;
+
+  //@ts-ignore because the `rawHeaders` property does actually exist
+  sanitizedRecording.rawHeaders = [];
```

Comment:
For some reason the type declarations in Nock is wrong. The `Definition` type should have the `rawHeaders` property.